### PR TITLE
Work in progress redesign

### PR DESCRIPTION
**Pull request by samdark**

### CODE REVIEW

File: assets/AppAsset.php [hunk 1]
```diff
@@ -18,7 +18,8 @@ class AppAsset extends AssetBundle
     public $basePath = '@webroot';
     public $baseUrl = '@web';
     public $css = [
-        'css/site.css',
+//        'css/site.css',
```
Comment (samdark) [inline assets/AppAsset.php]: Would need to be cleaned up before PR is merged.

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,9 @@
         "codemix/yii2-localeurls": "~1.4.10",
         "bower-asset/magnific-popup": "~1.1",
         "yiisoft/yii2-jui": "~2.0.0",
-        "bower-asset/cropper": "@dev"
+        "bower-asset/cropper": "@dev",
+        "bower-asset/font-awesome": "*",
+        "claviska/simpleimage": "^3.3"
```
Comment (samdark) [inline composer.json]: What do we need it for? We have imagine already.

File: config/web.php [hunk 3]
```diff
@@ -61,7 +95,7 @@
         ],
         'authClientCollection' => [
             'class' => 'yii\authclient\Collection',
-            'clients' => require __DIR__ . '/authclients.php',
+//            'clients' => require __DIR__ . '/authclients.php',
```
Comment (samdark) [inline config/web.php]: Needs to be back before merging this PR.

File: controllers/ProjectController.php [hunk 4]
```diff
@@ -73,9 +75,14 @@ public function actionIndex()
                 ->limit($limit)
         ]);
 
+        $projectsCount = (clone $newProvider->query)->limit(null)->count();
```
Comment (samdark) [inline controllers/ProjectController.php]: The number could be cached and invalidated when a project is added / deleted.

File: messages/ru/project.php [hunk 5]
```diff
@@ -29,14 +32,14 @@
...
-    'Description in {language}' => 'Описание на языке "{language}"',
+    'Description in {language}' => 'Описание на {language, select, Русский{русском} other{языке {language}}}',
     'Does not matter' => 'Не важно',
     'Draft' => 'Черновик',
     'Featured projects' => 'Выбор редакции',
     'General' => 'Общая информация',
     'ID' => 'ID',
     'Is Featured' => 'Выбор редакции',
-    'Is OpenSource' => 'OpenSource',
+    'Is OpenSource' => 'Является OpenSource',
```
Comment (samdark) [inline messages/ru/project.php]: Тут и без "является" всё понятно.

File: models/Image.php [hunk 6]
```diff
@@ -175,35 +210,56 @@ public function ensureThumbnailPath()
...
-        
-        if ($crop === null) {
-            ImagineImage::thumbnail($this->getOriginalPath(), $size[0], $size[1], ImageInterface::THUMBNAIL_INSET)
-                ->save($this->ensureThumbnailPath());    
-        } else {
-            ImagineImage::crop($this->getOriginalPath(), $crop['width'], $crop['height'], [$crop['x'], $crop['y']])
-                ->thumbnail(new Box($size[0], $size[1]), ImageInterface::THUMBNAIL_INSET)
-                ->save($this->ensureThumbnailPath());
+
+        $image = new SimpleImage($this->getOriginalPath());
```
Comment (samdark) [inline models/Image.php]: Why SimpleImage is better than imagine?

File: models/Image.php [hunk 7]
```diff
@@ -2,6 +2,8 @@
 
 namespace app\models;
 
+use claviska\SimpleImage;
+use Eventviva\ImageResize;
```
Comment (samdark) [inline models/Image.php]: Not used?

File: models/Project.php [hunk 8]
```diff
@@ -47,8 +47,8 @@ class Project extends \yii\db\ActiveRecord
     const STATUS_DRAFT = 10;
     const STATUS_PUBLISHED = 20;
 
-    const YII_VERSION_11 = '1.1';
-    const YII_VERSION_20 = '2.0';
+    const YII_VERSION_11 = 'Yii 1.1';
```
Comment (samdark) [inline models/Project.php]: It's better to revert it. In URL it looks like `yiiVersion=Yii+2.0` while previously it was `yiiVersion=2.0`.

File: models/Tag.php [hunk 9]
```diff
@@ -15,6 +15,7 @@
  * @property integer $type
  * @property integer $created_at
  * @property integer $updated_at
+ * @property mixed frequency
```
Comment (samdark) [inline models/Tag.php]: Should be `int $frequency`.

File: models/TagQuery.php [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+use creocoder\taggable\TaggableQueryBehavior;
+use yii\db\ActiveQuery;
+use yii\db\Expression;
+
+/**
+ * TagQuery
+ */
+class TagQuery extends ActiveQuery
+{
+    public function top($count = 10)
```
Comment (samdark) [inline models/TagQuery.php]: `$count` isn't used.

File: views/project/_card.php [hunk 11]
```diff
@@ -9,8 +9,17 @@
...
+<article class="<?= $model->getStatusClass() ?>">
+    <a class="details" href="<?= Url::to(['project/view', 'id' => $model->id, 'slug' => $model->slug]) ?>">
+        <img class="img-responsive" src="<?= $model->getPrimaryImageThumbnailRelativeUrl() ?>" />
+        <h1><?= Html::encode($model->title) ?></h1>
+    </a>
+
+    <p class="tags">
+        <?= implode(', ', array_map(function ($tag) {
+            return Html::a(Html::encode($tag->name), ['/project/list', 'tags' => $tag->name]);
+        }, array_slice($model->tags, 0, 10))); ?>
```
Comment (samdark) [inline views/project/_card.php]: It causes lots and lots of SQL queries. `with()` should be used when selecting data.

File: views/project/index.php [hunk 12]
```diff
@@ -1,36 +1,57 @@
...
-            'itemOptions' => ['class' => 'item'],
-            'itemView' => '_card'
+            'itemOptions' => ['class' => 'project'],
+            'itemView' => '_card',
         ]) ?>
-    </div>
 
-    <?= \yii\bootstrap\Html::a(Yii::t('project', 'More projects'), ['project/list'], ['class' => 'btn btn-default']) ?>
+        <p class="show-more">
+            <?= Html::a(Yii::t('project', 'View other {n, plural, one{# project} other{# projects}}', [
```
Comment (samdark) [inline views/project/index.php]: Check {n} more {n, plural, one{project} other{projects}}

File: web/css/style.css [hunk 13]
```diff
@@ -0,0 +1,2918 @@
...
+}
+input[type="search"] {
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+}
+input[type="radio"],
+input[type="checkbox"] {
+  margin: 4px 0 0;
+  margin-top: 1px \9;
```
Comment (samdark) [inline web/css/style.css]: `\9`?

File: web/css/style.css [hunk 14]
```diff
@@ -0,0 +1,2918 @@
...
+  padding: 0;
+  margin: 0;
+  border: 0;
+  min-width: 0;
+}
+legend {
+  display: block;
+  width: 100%;
+  padding: 0;
+  margin-bottom: 20px;
```
Comment (samdark) [inline web/css/style.css]: px sizes aren't good idea for mobile.

File: web/css/style.css [hunk 15]
```diff
@@ -0,0 +1,2918 @@
...
+input[type="radio"]:focus,
+input[type="checkbox"]:focus {
+  outline: 5px auto -webkit-focus-ring-color;
+  outline-offset: -2px;
+}
+output {
+  display: block;
+  padding-top: 7px;
+  font-size: 14px;
+  line-height: 1.42857143;
```
Comment (samdark) [inline web/css/style.css]: What's the magic number?

File: web/css/style.css [hunk 16]
```diff
@@ -0,0 +1,2918 @@
...
+  font-weight: normal;
+  cursor: pointer;
+}
+.radio input[type="radio"],
+.radio-inline input[type="radio"],
+.checkbox input[type="checkbox"],
+.checkbox-inline input[type="checkbox"] {
+  position: absolute;
+  margin-left: -20px;
+  margin-top: 4px \9;
```
Comment (samdark) [inline web/css/style.css]: `\9`?

File: web/css/style.css [hunk 17]
```diff
@@ -0,0 +1,2918 @@
...
+  color: #8a6d3b;
+}
+.has-warning .form-control {
+  border-color: #8a6d3b;
+  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
+  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
+}
+.has-warning .form-control:focus {
+  border-color: #66512c;
+  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075), 0 0 6px #c0a16b;
```
Comment (samdark) [inline web/css/style.css]: No need for webkit-specific rule: https://caniuse.com/#search=box-shadow

File: web/css/style.css [hunk 18]
```diff
@@ -0,0 +1,2918 @@
...
+.projects-list .projects:before,
+.projects-list .projects:after,
+.projects-list .projects .filters-wrapper .tags .list .item a:before,
+.projects-list .projects .filters-wrapper .tags .list .item a:after,
+.project-add .form-box:before,
+.project-add .form-box:after,
+.project-add .form-box .form-wrapper:before,
+.project-add .form-box .form-wrapper:after {
+  content: " ";
+  display: table;
```
Comment (samdark) [inline web/css/style.css]: Won't work well on mobile. Better to use flex grid.

File: web/css/style.css [hunk 19]
```diff
@@ -0,0 +1,2918 @@
...
+  display: none !important;
+}
+.show {
+  display: block !important;
+}
+.invisible {
+  visibility: hidden;
+}
+.text-hide {
+  font: 0/0 a;
```
Comment (samdark) [inline web/css/style.css]: ?

File: models/Image.php (hunk 6 above)
Comment (SilverFire) [inline models/Image.php]: It provides convenient API, especially for https://github.com/claviska/SimpleImage/blob/master/src/claviska/SimpleImage.php#L506

File: models/TagQuery.php (hunk 10 above)
Comment (SilverFire) [inline models/TagQuery.php]: fixed

File: views/project/_card.php (hunk 11 above)
Comment (SilverFire) [inline views/project/_card.php]: fixed

File: web/css/style.css (hunk 13 above)
Comment (SilverFire) [inline web/css/style.css]: compiled from less

File: web/css/style.css (hunk 15 above)
Comment (SilverFire) [inline web/css/style.css]: compiled from less

File: web/css/style.css (hunk 16 above)
Comment (SilverFire) [inline web/css/style.css]: compiled from less

File: web/css/style.css (hunk 17 above)
Comment (SilverFire) [inline web/css/style.css]: compiled from less

File: web/css/style.css (hunk 19 above)
Comment (SilverFire) [inline web/css/style.css]: compiled from less

File: web/css/style.css (hunk 13 above)
Comment (samdark) [inline web/css/style.css]: Weird :)

File: messages/ru/project.php (hunk 5 above)
Comment (SilverFire) [inline messages/ru/project.php]: Оно отображается на странице создания проекта и так выглядит логичнее

File: models/Tag.php (hunk 9 above)
Comment (SilverFire) [inline models/Tag.php]: fixed

File: web/css/style.css (hunk 14 above)
Comment (SilverFire) [inline web/css/style.css]: fixed

File: views/project/view.php [hunk 20]
```diff
@@ -53,103 +37,79 @@
...
-            <?php endif ?>
+                <p><?= Yii::t('project', 'Yii Version') ?>: <?= Html::encode($model->yii_version) ?></p>
+
+                <ul class="tags">
+                    <?php foreach ($model->tags as $tag): ?>
+                        <li><?= Html::a(Html::encode($tag->name), ['project/list', 'tags' => $tag->name]) ?></li>
+                    <?php endforeach ?>
+                </ul>
+
+                <?php if ($management !== false && $canManageProject) : ?>
```
Comment (samdark) [inline views/project/view.php]: It's not defined in case you're not logged in as admin.

**Comment by samdark** [comment]
Merged https://github.com/samdark/yiipowered/commit/13249d916a9a8a8aac5702189341df09615bcd85. Thanks!

**Pull request closed without merging**